### ISSUE TITLE
MAVLink secure link icon

### ISSUE THREAD

**Issue by LorenzMeier** [mavlink/qgroundcontrol#2605]
We would want to create a new icon to indicate a secure link, much like HTTPS in a web browser.

**Comment by LorenzMeier** [comment]
@biapalmeiro @cfelipesouza Essentially the equivalent of the padlock in the browser.

**Comment by cfelipesouza** [comment]
Where this icon will be placed??

**Comment by LorenzMeier** [comment]
In the status bar at the top to indicate the user wether he has a secure link.

**Comment by dogmaphobic** [comment]
I would put it up on top on the toolbar. Something like this (showing an insecure connection). Clicking/touching the icon would bring up a small pop up explaining to the user.

![screen shot 2016-01-14 at 9 17 19 am copy](https://cloud.githubusercontent.com/assets/749243/12326802/da27fd90-baa0-11e5-986f-6cb84fdca1ab.png)

**Comment by LorenzMeier** [comment]
It would be fantastic if it could be a MAVLink / drone specific style. So that by looking at the icon, you know its secure MAVLink (and not just some "lock").

**Comment by cfelipesouza** [comment]
What do you guys think?

![qgc_securelink_01](https://cloud.githubusercontent.com/assets/7512740/12393640/218db842-bdde-11e5-9fac-a6f4a3714758.png)
![qgc_securelink_02](https://cloud.githubusercontent.com/assets/7512740/12393641/218f9b76-bdde-11e5-8242-ece62c779e51.png)

**Comment by DonLakeFlyer** [comment]
The issue is that it is QuadCopter oriented. People get consfued when we do things like that and they are flying a Plane or piloting a Rover or a Sub.

**Comment by LorenzMeier** [comment]
Yes, second that. It should not include the airframe itself. And it should be visually nimble so that you recognise it, but at the same time doesn't eat a lot of screen real estate.

**Comment by biapalmeiro** [comment]
Just to understand. Is it to show the drone is connected or it is an extra information?

**Comment by LorenzMeier** [comment]
It's the equivalent of the padlock in your browser to indicate a secure connection.

**Comment by cfelipesouza** [comment]
I understand that the icon represents a secure connection, but for us here, the information about the drone connection is missing, like on Mission Planner, where it has a socket to represent it. And it is more important, what do you think? Do you guys have the intention to implement it? Maybe we can merge this information (connection + security) in one single icon.

Another question: the scenario where the drone is connected but the connection is not safe. Does it happen?

**Comment by LorenzMeier** [comment]
Compare the connected and unconnected toolbar. The connected state is obvious.

**Comment by biapalmeiro** [comment]
@LorenzMeier We can notice the difference, but maybe for new users it is not enough. Or maybe if the entire screen changed and redirect the user to the correct screen could solve this perception. What do you think?

**Comment by DonLakeFlyer** [comment]
There is quite a bit of UI change that happens when a vehicle connects:
- Toolbar changes radically
- Parameter loader progress bar goes across the top of the screen
- Vehicle shows up on map
- Setup View populates with buttons

We haven't had anyone confused as to when something was connected. Don't think we need to add something. Less is more. I also think that the lock icon or whatever shouldn't be on the main screen. Does that need to take up space the entire time the vehicle is connected. After you see it once it provides no additional information. Vehicle Summary would be a better place. Possibly including a bottom popup, that tells you once you are on a secure connection.

**Comment by biapalmeiro** [comment]
@DonLakeFlyer Agreed the Vehicle Summary is the best place for that. In this case we don't need an icon, once this screen shows only text information. Neither a pop up, because all the info could be displayed directly on this screen. It could be placed on Safety block.
